app/backend/src/couchers/models/moderation.py

```diff
@@ -0,0 +1,189 @@
+        Index("ix_moderation_states_object", object_type, object_id, unique=True),
+        # Covering index for visibility filtering - enables index-only scans in where_moderated_content_visible
+        Index("ix_moderation_states_id_visibility", id, visibility),
+    )
+
+    def __repr__(self) -> str:
+        return f"ModerationState(id={self.id}, type={self.object_type}, object_id={self.object_id}, visibility={self.visibility})"
+
+
+class ModerationQueueItem(Base):
+    """
+    Action items in the moderation queue
+
+    This table tracks what moderators need to review. Items remain in the queue
+    until they are resolved (linked to a ModerationLog entry).
+    """
+
+    __tablename__ = "moderation_queue"
+
+    id: Mapped[int] = mapped_column(
+        BigInteger, moderation_seq, primary_key=True, server_default=moderation_seq.next_value()
+    )
+    moderation_state_id: Mapped[int] = mapped_column(ForeignKey("moderation_states.id"), index=True)
+
+    time_created: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
+    trigger: Mapped[ModerationTrigger] = mapped_column(Enum(ModerationTrigger))
+    reason: Mapped[str] = mapped_column(String)
+
+    # When resolved, this links to the log entry that resolved it
+    resolved_by_log_id: Mapped[int | None] = mapped_column(ForeignKey("moderation_log.id"), nullable=True, index=True)
+
+    # The user who authored the content being moderated
+    item_author_user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), index=True)
+
+    # Relationships
+    moderation_state: Mapped["ModerationState"] = relationship("ModerationState")
+    author: Mapped["User"] = relationship("User", foreign_keys="ModerationQueueItem.item_author_user_id")
+
+    __table_args__ = (
+        # Fast lookup of unresolved items
+        Index(
+            "ix_moderation_queue_unresolved",
+            moderation_state_id,
+            time_created,
+            postgresql_where=resolved_by_log_id.is_(None),
+        ),
+    )
+
+    def __repr__(self) -> str:
+        return (
+            f"ModerationQueueItem(id={self.id}, trigger={self.trigger}, resolved={self.resolved_by_log_id is not None})"
+        )
+
+
+class ModerationLog(Base):
+    """
+    History of moderation actions
+
+    This table provides a complete audit trail of all moderation actions taken,
+    including who performed the action and what changed.
+    """
+
+    __tablename__ = "moderation_log"
+
+    id: Mapped[int] = mapped_column(
+        BigInteger, moderation_seq, primary_key=True, server_default=moderation_seq.next_value()
+    )
+    moderation_state_id: Mapped[int] = mapped_column(ForeignKey("moderation_states.id"), index=True)
+
+    time: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
+    action: Mapped[ModerationAction] = mapped_column(Enum(ModerationAction))
+    moderator_user_id: Mapped[int] = mapped_column(ForeignKey("users.id"))
```

Comment:
moderator_id?

Reply:
We use `_user_id` suffic pretty commonly for user id columns. I'm not sure it's completely universal.